android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java

```diff
@@ -397,9 +399,16 @@ public void failed(@NonNull RemoveErrorCode errorCode) {
      */
     @ReactMethod
     public void reScanAndLoadWifiList(final Promise promise) {
+        boolean wifiStartScan = wifi.startScan();
+        Log.d(TAG, "wifi start scan: " + wifiStartScan);
+        if(wifiStartScan == true){
         final WifiScanResultReceiver wifiScanResultReceiver = new WifiScanResultReceiver(wifi, promise);
         getReactApplicationContext().registerReceiver(wifiScanResultReceiver, new IntentFilter(WifiManager.SCAN_RESULTS_AVAILABLE_ACTION));
-        wifi.startScan();
+        }else{
+            Log.d(TAG, "Wifi scan rejected");
+            promise.resolve("Starting Android 9, it's only allowed to scan 4 times per 2 minuts in a foreground app.");
```

Comment:
"Starting Android 9, it's only allowed to scan 4 times per 2 minutes in a foreground app."